daml-foundations/daml-ghc/tests/MonoidTest.daml

```diff
@@ -0,0 +1,15 @@
+-- Copyright (c) 2019, Digital Asset (Switzerland) GmbH and/or its affiliates.
+-- All rights reserved.
+
+daml 1.2
+module MonoidTest where
+
+import DA.Monoid
+
+testSum = scenario do
+  assert $ Sum (1 : Int) <> Sum 2 == Sum (1 + 2)
+  assert $ mempty == Sum (aunit : Int)
+
+testProduct = scenario do
+  assert $ Product (2 : Int) <> Product 3 == Product (2 * 3)
+  assert $ mempty == Product (munit : Int)
```

Comment:
I wish we had doctests. Then we could remove these. :)

Reply:
I was hoping to find some time to implement doctests sometime soon :slightly_smiling_face: